src/OrbitGl/TimeGraph.cpp

```diff
@@ -437,63 +438,162 @@ void TimeGraph::SelectAndMakeVisible(const TimerInfo* timer_info) {
   track_container_->VerticallyMoveIntoView(*timer_info);
 }
 
+static bool ThreadMatches(const std::optional<uint32_t>& target_thread_id, const TimerInfo* timer) {
+  return !target_thread_id || *target_thread_id == timer->thread_id();
+}
+
+static void UpdatePreviousTimerAndGoalTime(const TimerInfo*& previous_timer, uint64_t& goal_time,
```

Comment:
I think we usually use pointers rather than references for this case (possibly changing an existing outside value).